consul/src/main/java/com/linecorp/armeria/client/consul/ConsulEndpointGroupBuilder.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.consul;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.net.URI;
+import java.time.Duration;
+
+import com.linecorp.armeria.common.SessionProtocol;
+import com.linecorp.armeria.internal.consul.ConsulClientBuilder;
+import com.linecorp.armeria.server.consul.ConsulUpdatingListenerBuilder;
+
+/**
+ * A builder class for {@link ConsulEndpointGroup}.
+ */
+public final class ConsulEndpointGroupBuilder extends ConsulClientBuilder {
+    private static final long DEFAULT_HEALTH_CHECK_INTERVAL_MILLIS = 10_000;
+
+    private final String serviceName;
+    private long registryFetchIntervalMillis = DEFAULT_HEALTH_CHECK_INTERVAL_MILLIS;
+    private boolean useHealthyEndpoints;
+
+    ConsulEndpointGroupBuilder(String serviceName) {
+        this.serviceName = requireNonNull(serviceName, "serviceName");
+    }
+
```

Comment:
I think we need a setter for the endpoint selection strategy.
https://github.com/line/armeria/blob/master/eureka/src/main/java/com/linecorp/armeria/client/eureka/EurekaEndpointGroupBuilder.java#L101

Reply:
Ok.

Reply:
You can copy from here: https://github.com/line/armeria/blob/c7edfb686cf3b9d926ca12f10d724f024afed6f2/core/src/main/java/com/linecorp/armeria/client/endpoint/dns/DnsEndpointGroupBuilder.java#L173